spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -347,7 +364,7 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       Short.MaxValue.toFloat,
       0.0f) ++
       Range(0, dataSize).map(_ => r.nextFloat())
-    withNulls(values).toDF("a")
+    castTest(withNulls(values).toDF("a"), DataTypes.StringType)
```

Comment:
Fixing a regression that was introduced in https://github.com/apache/datafusion-comet/pull/346